### PR TITLE
Adding ColPack coloring

### PR DESCRIPTION
**Pull request by michel2323**

### CODE REVIEW

File: src/autodiff.jl [hunk 1]
```diff
@@ -12,6 +15,78 @@ using ..ExaPF: State, Control
...
+
+export SparseDiffToolsColoring, ColPackColoring
+
+"""
+    matrix_colors(M::SparseMatrixCSC, ::SparseDiffToolsColoring)
+
+Calls the default coloring algorithm of `SparseDiffTools.jl`
+
+"""
+function matrix_colors(M::SparseMatrixCSC, ::SparseDiffToolsColoring)
```
Comment (frapac) [inline src/autodiff.jl:50]: Nice!

File: src/autodiff.jl [hunk 2]
```diff
@@ -12,6 +15,78 @@ using ..ExaPF: State, Control
...
+                push!(i,j[k])
+                push!(j,i[k])
+                push!(v,v[k])
+            end
+        end
+        # Values don't matter for coloring but for symmetry
+        v .= 1.0
+        M = sparse(i,j,v,size(M,1), size(M,2))
+    end
+    colpack_coloring = ColPack.ColPackColoring(M, d2_coloring(), natural_ordering())
```
Comment (frapac) [inline src/autodiff.jl:82]: `d2_coloring`? Or did you meant `d1_coloring`?

File: src/autodiff.jl [hunk 3]
```diff
@@ -12,6 +15,78 @@ using ..ExaPF: State, Control
...
+    return ColPack.get_colors(colpack_coloring)
+end
+
+function matrix_colors(M::SparseMatrixCSC; coloring::AbstractColoring=ColPackColoring())
```
Comment (frapac) [inline src/autodiff.jl:86]: Oups I think this is an anti-pattern. I would suggest removing this function.

File: test/Polar/matpower.jl [hunk 4]
```diff
@@ -41,68 +41,68 @@ const INSTANCES_DIR = joinpath(artifact"ExaData", "ExaData")
...
-
-    @test isapprox(stack.vmag, vmag_matpower, rtol=1e-6)
-end
+# @testset "Power flow 9 bus case (phase shift) (see Issue #184)" begin
```
Comment (frapac) [inline test/Polar/matpower.jl:44]: Arghhhh ... what happened there?

File: test/Polar/first_order.jl [hunk 5]
```diff
@@ -21,43 +21,49 @@ function test_constraints_jacobian(polar, device, MT)
...
-        Jmat = ExaPF.matpower_jacobian(polar, constraint, V)
-        Jmat = Jmat[:, mymap]
-
-        # Compare with FiniteDiff
-        function jac_fd_x(x)
-            stack.input[mymap] .= x
-            c = zeros(m) |> MT
-            constraint(c, stack)
-            return c
+        @testset "Colorings" for coloring in [
```
Comment (frapac) [inline test/Polar/first_order.jl:24]: nice!

File: src/Polar/second_order.jl [hunk 6]
```diff
@@ -84,14 +84,14 @@ struct FullHessian{Model, Func, VD, SMT, VI, Buff} <: AutoDiff.AbstractFullHessi
     H::SMT
 end
 
-function _get_hessian_colors(polar::PolarForm, func::AutoDiff.AbstractExpression, map::Vector{Int})
+function _get_hessian_colors(polar::PolarForm, func::AutoDiff.AbstractExpression, map::Vector{Int}, coloring::AutoDiff.AbstractColoring)
     H = _hessian_sparsity(polar, func)::SparseMatrixCSC
     Hsub = H[map, map] # reorder
-    colors = AutoDiff.SparseDiffTools.matrix_colors(Hsub)
+    colors = AutoDiff.matrix_colors(Hsub)
```
Comment (frapac) [inline src/Polar/second_order.jl:90]: We are not using the object `coloring` here right? is this expected?

**Comment by michel2323** [comment]
Converter back to draft, since the benefits of using ColPack in ExaPF are not yet clear.

**Pull request closed without merging**